File: buoi3/app/src/main/java/com/example/buoi3/MainActivity.java
package com.example.buoi3;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private EditText edtNumber;
    private EditText edtText;
    private CheckBox cbCheckBox;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        edtNumber = findViewById(R.id.edtNumber);
        cbCheckBox = findViewById(R.id.cbCheckBox);

        cbCheckBox.isChecked();
// lắng nghe sự kiện người sử dụng kich vào checkbox
        cbCheckBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
                if (compoundButton.isChecked()) {
                    Toast.makeText(MainActivity.this, "Bạn đã tích", Toast.LENGTH_SHORT).show();
                } else {
                    Toast.makeText(MainActivity.this, "Bạn không tích", Toast.LENGTH_SHORT).show();
                }
            }
        });

    }
}

File: buoi5/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    private var edt :EditText? = null
    private var edt1 :EditText? = null
    private var edt2 :EditText? = null
    private var bt :Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        edt = findViewById(R.id.edt)
        edt1 = findViewById(R.id.edt1)
        edt2 = findViewById(R.id.edt2)
        bt = findViewById(R.id.bt)
        bt?.setOnClickListener{
            val a: Double = edt?.text.toString().toDouble()
            val b: Double = edt1?.text.toString().toDouble()
            val c: Double = edt2?.text.toString().toDouble()
            val d = b*b -4*a*c
            if(d<0)
            {
                Toast.makeText(this,"vo nghiem", Toast.LENGTH_SHORT).show()
            }
        }
        }
}
